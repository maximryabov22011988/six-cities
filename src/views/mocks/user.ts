export default {
  avatar_url: '/static/avatar/8.jpg',
  email: 'Oliver.conner@gmail.com',
  id: 1,
  is_pro: false,
  name: 'Oliver.conner',
};
