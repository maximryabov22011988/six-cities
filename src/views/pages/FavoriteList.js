import React from 'react';
import PropTypes from 'prop-types';

import Page from '../components/Page';
import Header from '../components/Header';
import Logo from '../components/Logo';
import UserInfo from '../components/UserInfo';
import Content from '../components/Content';

import Footer from './favoriteList/Footer';

const propTypes = {
  isAuthUser: PropTypes.bool.isRequired,
  user: PropTypes.oneOfType([PropTypes.object, PropTypes.any]),
};

function FavoriteList({ isAuthUser, user }) {
  return (
    <Page className="page">
      <Header
        logo={<Logo position="header" />}
        userInfo={<UserInfo avatarUrl={user.avatar_url} email={user.email} isAuth={isAuthUser} />}
      />
      <Content page="favorites">
        <h1 className="favorites__title">Saved listing</h1>
        <ul className="favorites__list">
          <li className="favorites__locations-items">
            <div className="favorites__locations locations locations--current">
              <div className="locations__item">
                <a className="locations__item-link" href="#">
                  <span>Amsterdam</span>
                </a>
              </div>
            </div>
            <div className="favorites__places">
              <article className="favorites__card place-card">
                <div className="favorites__image-wrapper place-card__image-wrapper">
                  <a href="#">
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="110"
                      src="img/apartment-small-03.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div className="favorites__card-info place-card__info">
                  <div className="place-card__price-wrapper">
                    <div className="place-card__price">
                      <b className="place-card__price-value">&euro;180</b>
                      <span className="place-card__price-text">&#47;&nbsp;night</span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg className="place-card__bookmark-icon" height="19" width="18">
                        <use xlinkHref="#icon-bookmark" />
                      </svg>
                      <span className="visually-hidden">In bookmarks</span>
                    </button>
                  </div>
                  <div className="place-card__rating rating">
                    <div className="place-card__stars rating__stars">
                      <span style={{ width: '100%' }} />
                      <span className="visually-hidden">Rating</span>
                    </div>
                  </div>
                  <h2 className="place-card__name">
                    <a href="#">Nice, cozy, warm big bed apartment</a>
                  </h2>
                  <p className="place-card__type">Apartment</p>
                </div>
              </article>
              <article className="favorites__card place-card">
                <div className="favorites__image-wrapper place-card__image-wrapper">
                  <a href="#">
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="110"
                      src="img/room-small.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div className="favorites__card-info place-card__info">
                  <div className="place-card__price-wrapper">
                    <div className="place-card__price">
                      <b className="place-card__price-value">&euro;80</b>
                      <span className="place-card__price-text">&#47;&nbsp;night</span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg className="place-card__bookmark-icon" height="19" width="18">
                        <use xlinkHref="#icon-bookmark" />
                      </svg>
                      <span className="visually-hidden">In bookmarks</span>
                    </button>
                  </div>
                  <div className="place-card__rating rating">
                    <div className="place-card__stars rating__stars">
                      <span style={{ width: '80%' }} />
                      <span className="visually-hidden">Rating</span>
                    </div>
                  </div>
                  <h2 className="place-card__name">
                    <a href="#">Wood and stone place</a>
                  </h2>
                  <p className="place-card__type">Private room</p>
                </div>
              </article>
            </div>
          </li>
          <li className="favorites__locations-items">
            <div className="favorites__locations locations locations--current">
              <div className="locations__item">
                <a className="locations__item-link" href="#">
                  <span>Cologne</span>
                </a>
              </div>
            </div>
            <div className="favorites__places">
              <article className="favorites__card place-card">
                <div className="favorites__image-wrapper place-card__image-wrapper">
                  <a href="#">
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="110"
                      src="img/apartment-small-04.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div className="favorites__card-info place-card__info">
                  <div className="place-card__price-wrapper">
                    <div className="place-card__price">
                      <b className="place-card__price-value">&euro;180</b>
                      <span className="place-card__price-text">&#47;&nbsp;night</span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg className="place-card__bookmark-icon" height="19" width="18">
                        <use xlinkHref="#icon-bookmark" />
                      </svg>
                      <span className="visually-hidden">In bookmarks</span>
                    </button>
                  </div>
                  <div className="place-card__rating rating">
                    <div className="place-card__stars rating__stars">
                      <span style={{ width: '100%' }} />
                      <span className="visually-hidden">Rating</span>
                    </div>
                  </div>
                  <h2 className="place-card__name">
                    <a href="#">White castle</a>
                  </h2>
                  <p className="place-card__type">Apartment</p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </Content>
      <Footer className="footer container">
        <Logo height="33" position="footer" width="64" />
      </Footer>
    </Page>
  );
}

FavoriteList.propTypes = propTypes;

export default FavoriteList;
