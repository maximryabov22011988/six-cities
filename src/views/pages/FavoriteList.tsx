import * as React from 'react';

import { keys } from 'lodash';

import Page from '../components/Page';
import Header from '../components/Header';
import Logo from '../components/Logo';
import UserInfo from '../components/UserInfo';
import Content from '../components/Content';
import PlaceCardList from '../components/PlaceCardList';
import Footer from '../components/Footer';

import { User, onAddToFavorities, onRemoveFromFavorities } from '../interfaces';

interface Props {
  isAuthUser: boolean;
  isEmpty: boolean;
  offers: object;
  user: User;
  onAddOfferToFavorities: onAddToFavorities;
  onRemoveOfferFromFavorities: onRemoveFromFavorities;
}

class FavoriteList extends React.Component<Props> {
  handleAddToFavorities = (hotelId) => () => {
    const { onAddOfferToFavorities } = this.props;
    onAddOfferToFavorities(hotelId);
  };

  handleRemoveFromFavorities = (hotelId) => () => {
    const { onRemoveOfferFromFavorities } = this.props;
    onRemoveOfferFromFavorities(hotelId);
  };

  render() {
    const { offers, isAuthUser, isEmpty, user } = this.props;
    const cities = keys(offers);

    return (
      <Page className="page">
        <Header
          logo={<Logo position="header" />}
          userInfo={<UserInfo avatarUrl={user.avatar_url} email={user.email} isAuth={isAuthUser} />}
        />
        <Content isEmpty={isEmpty} parentClassName="favorites">
          {isEmpty ? (
            <React.Fragment>
              <h1 className="visually-hidden">Favorites (empty)</h1>
              <div className="favorites__status-wrapper">
                <b className="favorites__status">Nothing yet saved.</b>
                <p className="favorites__status-description">
                  Save properties to narrow down search or plan yor future trips.
                </p>
              </div>
            </React.Fragment>
          ) : (
            <React.Fragment>
              <h1 className="favorites__title">Saved listing</h1>
              <ul className="favorites__list">
                {cities.map((city) => (
                  <li className="favorites__locations-items" key={city}>
                    <div className="favorites__locations locations locations--current">
                      <div className="locations__item">
                        <span className="locations__item-link">
                          <span>{city}</span>
                        </span>
                      </div>
                    </div>
                    <PlaceCardList
                      offers={offers[city]}
                      parentClassName="favorites"
                      onAddToFavorities={this.handleAddToFavorities}
                      onRemoveFromFavorities={this.handleRemoveFromFavorities}
                    />
                  </li>
                ))}
              </ul>
            </React.Fragment>
          )}
        </Content>
        <Footer className="footer container">
          <Logo height="33" position="footer" width="64" />
        </Footer>
      </Page>
    );
  }
}

export default FavoriteList;
