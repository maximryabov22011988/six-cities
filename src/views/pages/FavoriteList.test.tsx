import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import FavoriteList from './FavoriteList';
import offer from '../mocks/offer';
import user from '../mocks/user';

const offers = {
  [offer.city.name]: [offer],
};

describe('FavoriteList', () => {
  it('renders correctly with offers', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <FavoriteList
            isAuthUser={true}
            isEmpty={false}
            offers={offers}
            user={user}
            onAddOfferToFavorities={jest.fn()}
            onRemoveOfferFromFavorities={jest.fn()}
          />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without offers', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <FavoriteList
            isAuthUser={true}
            isEmpty={true}
            offers={[]}
            user={user}
            onAddOfferToFavorities={jest.fn()}
            onRemoveOfferFromFavorities={jest.fn()}
          />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
