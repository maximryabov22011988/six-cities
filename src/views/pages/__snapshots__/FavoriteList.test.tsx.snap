// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteList renders correctly with offers 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper false"
                >
                  <img
                    alt="User avatar"
                    className="reviews__avatar user__avatar"
                    src="https://es31-server.appspot.com/six-cities/static/avatar/8.jpg"
                  />
                </div>
                <span
                  className="header__user-name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="container page__favorites-container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <span
                  className="locations__item-link"
                >
                  <span>
                    Paris
                  </span>
                </span>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="place-card favorites__card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <div
                      style={
                        Object {
                          "alignItems": "center",
                          "background": "#e1e1e1",
                          "borderRadius": "4px",
                          "display": "flex",
                          "height": "110px",
                          "justifyContent": "center",
                          "width": "150px",
                        }
                      }
                    >
                      <img
                        alt="Place image"
                        className="place-card__image"
                        height={0}
                        onError={[Function]}
                        onLoad={[Function]}
                        src="https://es31-server.appspot.com/six-cities/static/hotel/17.jpg"
                        width={0}
                      />
                      <span>
                        Loading ...
                      </span>
                    </div>
                  </a>
                </div>
                <div
                  className="place-card__info favorites__card-info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        161
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "54.00%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="/offer/1"
                      onClick={[Function]}
                    >
                      Canal View Prinsengracht
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    house
                  </p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;

exports[`FavoriteList renders correctly without offers 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper false"
                >
                  <img
                    alt="User avatar"
                    className="reviews__avatar user__avatar"
                    src="https://es31-server.appspot.com/six-cities/static/avatar/8.jpg"
                  />
                </div>
                <span
                  className="header__user-name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites page__main--favorites-empty"
  >
    <div
      className="container page__favorites-container"
    >
      <section
        className="favorites favorites--empty"
      >
        <h1
          className="visually-hidden"
        >
          Favorites (empty)
        </h1>
        <div
          className="favorites__status-wrapper"
        >
          <b
            className="favorites__status"
          >
            Nothing yet saved.
          </b>
          <p
            className="favorites__status-description"
          >
            Save properties to narrow down search or plan yor future trips.
          </p>
        </div>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
